Use authenticated GitHub API calls in linux bundle creating dockerfile

## Problem Definition

The automatic bundling for linux often fails with the following error on Travis:
```
Step 11/21 : ADD https://api.github.com/repos/${REPO}/commits/${RAIDENVERSION} /dev/null
ADD failed: failed to GET https://api.github.com/repos/raiden-network/raiden/commits/v0.4.2 with status 403 Forbidden: {"message":"API rate limit exceeded for 35.202.145.110. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)","documentation_url":"https://developer.github.com/v3/#rate-limiting"}
make: *** [bundle-docker] Error 1
```

The reason appears to be that too many unauthenticated requests to the github API are performed form Travis' IP addresses.
To prevent this we should be using authenticated requests.

--- a/.travis.yml
+++ b/.travis.yml
@@ -5,7 +5,7 @@ stages:
   - name: integration
     if: type = cron OR commit_message =~ /\[ci integration\]/ OR tag IS present
   - name: deploy
-    if: tag IS present AND tag =~ /^v[\d]+\.[\d]+\.[\d]+/
+    if: type = cron OR tag IS present AND tag =~ /^v[\d]+\.[\d]+\.[\d]+/
 
 templates:
   job-template-test: &job-template-test
@@ -42,6 +42,9 @@ env:
   - SYNAPSE_URL='https://github.com/matrix-org/synapse/tarball/master#egg=synapse'
   - SYNAPSE_SERVER_NAME='matrix.local.raiden'
   - COVERALLS_PARALLEL=true
+  # `GITHUB_ACCESS_TOKEN` - read only access token to increase GitHub API rate limits
+  # Format: `GITHUB_ACCESS_TOKEN=<username>:<token>`
+  - secure: "riRxis2NVGAjgl3jXOnMCXCpfWhOK7Q5B7CCRC0xPk+CyuHEOjIAvmueZwCrWt6i4jjlPeqpsLzDp7j56QG94hfGAs1sBq7aZQ6qVSqjCsj5IYPdAYRq8nivgHWjQuPoDWKsKOcEw44lUXUrWXvOf9T+AM+numz/feISdchqFr+OhVYv0g8mDQlFwAMcIwu4DpazECV/LlQxFR3iP7GDF6tU7NDx5uzCYzRU2vg4sSZWJJLHLs9gCpAqlEiKL3n5QhonQwJ7qbgOTlJaJNrI1qhOw29LjcVqIMxkM1VJ1hx61e55WzM/OquijMJCNrNVCQ8m3RP//bxoGoMrj9FFI+8qT9TSPjvSXWyZQxak30zZE7szF1Zfu/Cd9f56D/28leQ+KuJh8SvK53x04eNxn0y+amSElHlL1NgOJ86XNTdZM6xcYGEbt6Fy0Ia5CyGkUUw+gsLVAjSghC/iQsZgmXyjSsDCRBpYO79mR2r3S240Zb1DIhi9OX5W1IO6DOGlzjOdrRehmMZ2T4trM+Rz0q06g9D1Wxn69IP/wYACgydO0B55BjD6bEtspot/yh7aHH8MFEqeYLpskXwg4CjZ6HMfccfo/KM3kElVQpFQO3JCguhCX8zb3P4o49pInr6icDMBQfnDWkSDi1MCY8GnbTyB9dL1T9J7bTje7vaCvdk="
 cache:
   pip: true
   directories:
@@ -159,20 +162,26 @@ jobs:
       before_install: mkdir -p $HOME/.bin; export PATH=$PATH:$HOME/.bin; .travis/before_install.sh
       install: .travis/install.sh
       script:
-        - RAIDENVERSION=$TRAVIS_TAG make bundle-docker
+        - source .travis/set_tag.sh
+        - REPO=${TRAVIS_REPO_SLUG} RAIDENVERSION=${TRAVIS_COMMIT} make bundle-docker
+        - tar -C dist -xvzf dist/archive/raiden-${ARCHIVE_TAG}-linux.tar.gz
         # Make sure the produced binary works
-        - dist/raiden-${TRAVIS_TAG}-linux smoketest
+        # Skip smoketest until raiden-network/raiden#2384 is fixed
+        #- dist/raiden-${ARCHIVE_TAG}-macOS smoketest
+
       deploy:
       - provider: releases
         api_key:
           secure: O85yltVQ1OacJQoNQWDJqOsXuSb+x7IMJTtUPuFM5iAOIGLmzhk0/qzJInixCGbAfWMYT4wTjA6OfxeiAWYzC5pCgqedZOsjp+8bXjn2Ek443Y+FBay3g38ebn3KBFBat3iMOamdrFSGTENugdG418GdK6yFVYpyCCUJXWLBcBUPOi7BlakI8TkznsaCad7OVRrPfMHWRe11WI3b4fajPnH/M7JRdDpT5GTisEchwKzYHnBttYySDZcQS3gtAEK7Srg9AkdEYCW3DOkmW4DeEgrQAQVcHHAV7eZREC1M3SrsrCotEqhEmIy29akOVKpOZtYWMxNFIsScDt39Y2hJLXRo1NjN0RBgIOg+Asl3B1cuK3Zfzc30r+Fyfkl9yFXpC35xddh2dHz5PFyssmtVZMEdxECtQT0DN08FB1JNdBbz3tfNw84IvR2ymYcQdRFWQM51rBUjUGspzDftAdITKkk3Z67jgYnaVvL0iBMbuuFjZDs8375IjIo7hxJ5UY7YWLV8Ic2r89wq5Dw7EsL5cyc4BVfj1kByZ1BJ6sZEKaqVdDdEc1Bv2/SYk6Da/muPB9gYK8Baw1gumLvrVS65EUVvwGmnq0zKrKmvTLl0aNPRNInaOVOloK63HdOwJhiLRn50YfCARrX+5kKBIQxdtNtmfPmdYJ3cm5bNhaXQ8fs=
-        file: dist/raiden-${TRAVIS_TAG}-linux.tar.gz
+        file: dist/archive/raiden-*-linux.tar.gz
+        file_glob: true
         skip_cleanup: true
         on:
           tags: true
           repo: raiden-network/raiden
           branch: master
-# Disabled until automated release account is reactivated
+
+## Disabled until automated release account is reactivated
 #      - provider: pypi
 #        user: raiden_pypi_automated
 #        password:
@@ -184,6 +193,23 @@ jobs:
 #          repo: raiden-network/raiden
 #          branch: master
 
+      - provider: s3
+        # Remove once https://github.com/travis-ci/dpl/issues/865 is fixed
+        edge:
+          branch: ha-s3-endpoint
+        skip_cleanup: true
+        access_key_id: MK22XIZDUYA3EWBHGAKD
+        secret_access_key:
+          secure: "VpTNi2O1kDyMfGCyZ4dRonmwp1cPl99ry/I34EdXg2qk8EL8d3Z7pVD0RwJjHGNTCZN0SYqj+MlbLQzBR0Ydxtpze5ayqN+v4aqvpdu33Vqm3gMJIgT6ciiGuBl6XwT3hhb6hWwSORXaGTYTn5G6fYPsd78A9/24YORf8kn9w5iBbjl1mBNTBw7e+SARTG2fHK0cM4qSNRMELfZvno7xcu2pjZK4vplOnSIxxE6Eh80PDE/AKyNUkRcHQ81Xt5QBekMwmalQUYRoZx75VYLSmrvCu1t407pfb80vxFaH+UG0mzuyUmZ/G8S6WY58jMBvv6EULr4Lj+ztT/CmnVPWlcLNP8hTJ+wBtdzgOVNFAJtL+PuudkMgE/JLuuF+IiRKjcIhBL29hGm46Kow3u6oJ4NvaTm0+8Vt+435TveHDg6bDwOl42ik66tZuklYmLadTd/SfxkUHdmIWsVGnfprwSmYrrkoPww0N90x3bbdfOtCI2mFeA4MThnkbOpmX9HOOOEYP0d3zTvu03h8P0N0nTsNiOyntMZ7943oJhlMEpGnhd/eHRAuYwO6s8mJZeQk7TCPvgbOnhGdtqR5SGLy3Q5+JV0sAieZeZvAbl+XIqDA0SPuJ4Jq/wCHaQ4B6fIcvW1PWl9gMleo5/Zh+yLXXuBKBPCTyOzaz97bNRQ/meM="
+        bucket: "raiden-nightlies"
+        acl: public_read
+        local_dir: dist/archive
+        cache_control: "max-age=31536000"
+        endpoint: https://ams3.digitaloceanspaces.com
+        on:
+          repo: raiden-network/raiden
+          branch: master
+
     - stage: deploy
       os: osx
       language: generic
@@ -191,29 +217,32 @@ jobs:
       env: TEST=""
       before_install:
         - sysctl -n machdep.cpu.brand_string
-        - git rev-parse HEAD
         - mkdir -p $HOME/.bin
-        - export PATH=$HOME/.bin:$HOME/Library/Python/3.7/bin:/usr/local/bin/:$PATH
+        - export PATH=$HOME/.bin:$HOME/Library/Python/3.6/bin:/usr/local/bin/:$PATH
         - .travis/before_install.sh
       install:
+        - git pull --tags
         - .travis/install.sh
+        # pdbpp interferes with pyinstaller, uninstall it
+        - pip uninstall -y pdbpp
       before_script:
         - python setup.py build
         - RAIDEN_NPM_MISSING_FATAL=1 python setup.py compile_webui
       script:
-        # pdbpp interferes with pyinstaller, uninstall it
-        - pip uninstall -y pdbpp
+        - source .travis/set_tag.sh
+        - mkdir -p dist/archive
         - pyinstaller --noconfirm --clean raiden.spec
-        - cd dist && zip -9 -r raiden-${TRAVIS_TAG}-macOS.zip raiden*
-        # Make sure the produced binary works
-        - dist/raiden-${TRAVIS_TAG}-macOS smoketest
+        - zip -9 --junk-paths dist/archive/raiden-${ARCHIVE_TAG}-macOS.zip dist/raiden*
+        # Skip smoketest for now, because smoketest can't use anything from raiden.tests
+        #- dist/raiden-${ARCHIVE_TAG}-macOS smoketest
         # "Fix" Travis issues #6307 #6522 #9486
         - shell_session_update() { :; };
       deploy:
         - provider: releases
           api_key:
             secure: O85yltVQ1OacJQoNQWDJqOsXuSb+x7IMJTtUPuFM5iAOIGLmzhk0/qzJInixCGbAfWMYT4wTjA6OfxeiAWYzC5pCgqedZOsjp+8bXjn2Ek443Y+FBay3g38ebn3KBFBat3iMOamdrFSGTENugdG418GdK6yFVYpyCCUJXWLBcBUPOi7BlakI8TkznsaCad7OVRrPfMHWRe11WI3b4fajPnH/M7JRdDpT5GTisEchwKzYHnBttYySDZcQS3gtAEK7Srg9AkdEYCW3DOkmW4DeEgrQAQVcHHAV7eZREC1M3SrsrCotEqhEmIy29akOVKpOZtYWMxNFIsScDt39Y2hJLXRo1NjN0RBgIOg+Asl3B1cuK3Zfzc30r+Fyfkl9yFXpC35xddh2dHz5PFyssmtVZMEdxECtQT0DN08FB1JNdBbz3tfNw84IvR2ymYcQdRFWQM51rBUjUGspzDftAdITKkk3Z67jgYnaVvL0iBMbuuFjZDs8375IjIo7hxJ5UY7YWLV8Ic2r89wq5Dw7EsL5cyc4BVfj1kByZ1BJ6sZEKaqVdDdEc1Bv2/SYk6Da/muPB9gYK8Baw1gumLvrVS65EUVvwGmnq0zKrKmvTLl0aNPRNInaOVOloK63HdOwJhiLRn50YfCARrX+5kKBIQxdtNtmfPmdYJ3cm5bNhaXQ8fs=
-          file: dist/raiden-${TRAVIS_TAG}-macOS.zip
+          file: dist/archive/raiden-*-macOS.zip
+          file_glob: true
           skip_cleanup: true
           on:
             tags: true
@@ -224,5 +253,21 @@ jobs:
           skip_cleanup: true
           on:
             tags: true
+            repo: raiden-network/raiden
+            branch: master
+        - provider: s3
+          # Remove once https://github.com/travis-ci/dpl/issues/865 is fixed
+          edge:
+            branch: ha-s3-endpoint
+          skip_cleanup: true
+          access_key_id: MK22XIZDUYA3EWBHGAKD
+          secret_access_key:
+            secure: "VpTNi2O1kDyMfGCyZ4dRonmwp1cPl99ry/I34EdXg2qk8EL8d3Z7pVD0RwJjHGNTCZN0SYqj+MlbLQzBR0Ydxtpze5ayqN+v4aqvpdu33Vqm3gMJIgT6ciiGuBl6XwT3hhb6hWwSORXaGTYTn5G6fYPsd78A9/24YORf8kn9w5iBbjl1mBNTBw7e+SARTG2fHK0cM4qSNRMELfZvno7xcu2pjZK4vplOnSIxxE6Eh80PDE/AKyNUkRcHQ81Xt5QBekMwmalQUYRoZx75VYLSmrvCu1t407pfb80vxFaH+UG0mzuyUmZ/G8S6WY58jMBvv6EULr4Lj+ztT/CmnVPWlcLNP8hTJ+wBtdzgOVNFAJtL+PuudkMgE/JLuuF+IiRKjcIhBL29hGm46Kow3u6oJ4NvaTm0+8Vt+435TveHDg6bDwOl42ik66tZuklYmLadTd/SfxkUHdmIWsVGnfprwSmYrrkoPww0N90x3bbdfOtCI2mFeA4MThnkbOpmX9HOOOEYP0d3zTvu03h8P0N0nTsNiOyntMZ7943oJhlMEpGnhd/eHRAuYwO6s8mJZeQk7TCPvgbOnhGdtqR5SGLy3Q5+JV0sAieZeZvAbl+XIqDA0SPuJ4Jq/wCHaQ4B6fIcvW1PWl9gMleo5/Zh+yLXXuBKBPCTyOzaz97bNRQ/meM="
+          bucket: "raiden-nightlies"
+          acl: public_read
+          local_dir: dist/archive
+          cache_control: "max-age=31536000"
+          endpoint: https://ams3.digitaloceanspaces.com
+          on:
             repo: raiden-network/raiden
             branch: master

--- a/.travis/prepare_os_osx.sh
+++ b/.travis/prepare_os_osx.sh
@@ -3,7 +3,12 @@
 set -e
 set -x
 
-for tool in automake gmp leveldb libffi libtool node openssl pkg-config python3; do
+PYTHON_36_FORMULA=https://raw.githubusercontent.com/Homebrew/homebrew-core/f2a764ef944b1080be64bd88dca9a1d80130c558/Formula/python.rb
+
+# Pin to python3.6 for the time being
+brew install ${PYTHON_36_FORMULA} || brew upgrade ${PYTHON_36_FORMULA}
+
+for tool in automake gmp leveldb libffi libtool node openssl pkg-config; do
     brew install ${tool} || brew upgrade ${tool} || true
 done
 

--- a/.travis/set_tag.sh
+++ b/.travis/set_tag.sh
@@ -1,0 +1,7 @@
+#!/bin/bash
+
+if [[ ! -z ${TRAVIS_TAG} ]]; then
+    export ARCHIVE_TAG=${TRAVIS_TAG}
+else
+    DATE=$(date +%Y.%m.%d) && export ARCHIVE_TAG="nightly-$DATE"
+fi

--- a/.travis/update_formula.sh
+++ b/.travis/update_formula.sh
@@ -16,7 +16,7 @@ setup_git() {
 }
 
 update_formula() {
-    UPDATED_SHA256=$(openssl sha -sha256 dist/raiden-${TRAVIS_TAG}-macOS.zip)
+    UPDATED_SHA256=$(openssl sha -sha256 dist/archive/raiden-${TRAVIS_TAG}-macOS.zip)
     FORMULA_FILE="homebrew-raiden/raiden.rb"
 
     sed -i .bak "s/[0-9]\.[0-9]\.[0-9]/${TRAVIS_TAG/v/}/g" $FORMULA_FILE

--- a/Makefile
+++ b/Makefile
@@ -62,13 +62,21 @@ docs:
 
 REPO?=raiden-network/raiden
 RAIDENVERSION?=master
+ARCHIVE_TAG?=$(RAIDENVERSION)
+
+GITHUB_ACCESS_TOKEN_ARG=
+ifdef GITHUB_ACCESS_TOKEN
+GITHUB_ACCESS_TOKEN_ARG=--build-arg GITHUB_ACCESS_TOKEN_FRAGMENT=$(GITHUB_ACCESS_TOKEN)@
+endif
+
 
 bundle-docker:
-	docker build -t pyinstallerbuilder --build-arg GETH_URL_LINUX=$(GETH_URL_LINUX) --build-arg SOLC_URL_LINUX=$(SOLC_URL_LINUX) --build-arg RAIDENVERSION=$(RAIDENVERSION) --build-arg REPO=$(REPO) -f docker/build.Dockerfile docker
+    # Hide command echo to prevent leaking GITHUB_ACCESS_TOKEN in Travis logs
+	@docker build -t pyinstallerbuilder --build-arg GETH_URL_LINUX=$(GETH_URL_LINUX) --build-arg SOLC_URL_LINUX=$(SOLC_URL_LINUX) --build-arg RAIDENVERSION=$(RAIDENVERSION) --build-arg REPO=$(REPO) --build-arg ARCHIVE_TAG=$(ARCHIVE_TAG) $(GITHUB_ACCESS_TOKEN_ARG) -f docker/build.Dockerfile docker
 	-(docker rm builder)
 	docker create --name builder pyinstallerbuilder
-	mkdir -p dist
-	docker cp builder:/raiden/raiden-$(RAIDENVERSION)-linux.tar.gz dist/raiden-$(RAIDENVERSION)-linux.tar.gz
+	mkdir -p dist/archive
+	docker cp builder:/raiden/raiden-$(ARCHIVE_TAG)-linux.tar.gz dist/archive/raiden-$(ARCHIVE_TAG)-linux.tar.gz
 	docker rm builder
 
 bundle:

--- a/docker/build.Dockerfile
+++ b/docker/build.Dockerfile
@@ -28,12 +28,17 @@ RUN cd /tmp && \
 # use --build-arg RAIDENVERSION=v0.0.3 to build a specific (tagged) version
 ARG REPO=raiden-network/raiden
 ARG RAIDENVERSION=master
+ARG GITHUB_ACCESS_TOKEN_FRAGMENT
 
 # This is a "hack" to automatically invalidate the cache in case there are new commits
-ADD https://api.github.com/repos/${REPO}/commits/${RAIDENVERSION} /dev/null
+ADD https://${GITHUB_ACCESS_TOKEN_FRAGMENT}api.github.com/repos/${REPO}/commits/${RAIDENVERSION} /dev/null
 
 # clone raiden repo + install dependencies
-RUN git clone -b ${RAIDENVERSION} https://github.com/${REPO}
+RUN git clone https://github.com/${REPO} && \
+    cd raiden && \
+    git fetch --tags && \
+    git reset --hard ${RAIDENVERSION}
+
 WORKDIR /raiden
 RUN git fetch --tags
 RUN pip install -r requirements.txt
@@ -54,7 +59,9 @@ RUN pip install pyinstaller
 # build pyinstaller package
 RUN pyinstaller --noconfirm --clean raiden.spec
 
+ARG ARCHIVE_TAG=${RAIDENVERSION}
+
 # pack result to have a unique name to get it out of the container later
 RUN cd dist && \
-    tar -cvzf raiden-${RAIDENVERSION}-linux.tar.gz raiden* && \
-    mv raiden-${RAIDENVERSION}-linux.tar.gz ..
+    tar -cvzf raiden-${ARCHIVE_TAG}-linux.tar.gz raiden* && \
+    mv raiden-${ARCHIVE_TAG}-linux.tar.gz ..

--- a/raiden/log_config.py
+++ b/raiden/log_config.py
@@ -269,7 +269,7 @@ def configure_logging(
     # if py-evm is not used this will throw, hence the try-catch block
     # for some reason it didn't work to put this into conftest.py
     try:
-        from eth.utils.logging import setup_trace_logging
+        from eth.tools.logging import setup_trace_logging
         setup_trace_logging()
-    except ModuleNotFoundError:
+    except ImportError:
         pass

--- a/raiden/tests/utils/events.py
+++ b/raiden/tests/utils/events.py
@@ -1,4 +1,5 @@
-from web3.utils.datastructures import AttributeDict
+from web3.datastructures import AttributeDict
+
 
 NOVALUE = object()
 

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -27,5 +27,3 @@ releases==1.6.1
 
 # Release
 bump2version==0.5.8
-
-eth-tester[py-evm]==0.1.0b31

--- a/requirements.txt
+++ b/requirements.txt
@@ -16,14 +16,15 @@ netifaces==0.10.7
 networkx==2.1.0
 psutil==5.4.5
 git+https://github.com/raiden-network/raiden-libs@c0afdfdd200eedcc288ddad277549a342c4c8146#egg=raiden-libs
-git+https://github.com/raiden-network/raiden-contracts@d77e2a7f1f83afd302caabf1179374879bc7725c#egg=raiden-contracts
+git+https://github.com/raiden-network/raiden-contracts@ce3fad4271c6b9a074d226509bd7a1f62d873075#egg=raiden-contracts
 webargs==1.8.1
 eth-keyfile==0.5.1
 eth-utils==1.2.0
 structlog
 colorama
 py-solc
-web3==4.6.0
+# Use master until https://github.com/ethereum/web3.py/issues/1048 is fixed
+git+https://github.com/ethereum/web3.py@e39518638682d02314e3f4700dea48ae95abf819#egg=web3[tester]
 pysha3==1.0.2
 gevent==1.3.6
 pytoml==0.1.18

